core/chains/evm/txmgr/builder.go

```diff
@@ -91,6 +94,56 @@ func NewEvmTxm(
 	return txmgr.NewTxm(chainId, cfg, txCfg, keyStore, lggr, checkerFactory, fwdMgr, txAttemptBuilder, txStore, broadcaster, confirmer, resender, tracker, finalizer, client.NewTxError)
 }
 
+func NewTxmV2(
+	ds sqlutil.DataSource,
+	chainConfig ChainConfig,
+	fCfg FeeConfig,
+	txConfig config.Transactions,
+	txmV2Config config.TxmV2,
+	client client.Client,
+	lggr logger.Logger,
+	logPoller logpoller.LogPoller,
+	keyStore keystore.Eth,
+	estimator gas.EvmFeeEstimator,
+) (TxManager, error) {
+	var fwdMgr *forwarders.FwdMgr
+	if txConfig.ForwardersEnabled() {
+		fwdMgr = forwarders.NewFwdMgr(ds, client, logPoller, lggr, chainConfig)
+	} else {
+		lggr.Info("ForwarderManager: Disabled")
+	}
+
+	chainID := client.ConfiguredChainID()
+
+	var stuckTxDetector txm.StuckTxDetector
```

Comment:
Any reason you have all the extra code around StuckTxDetector to check if it is enabled or not?
Just adds more complexity here, and in the txm.
In the Txmv1, all that logic is inside the StuckTxDetector, and that makes the txm code cleaner.
Can't we do the same pattern here too?